src/ipa_pc/data_structures.rs

```diff
@@ -51,83 +107,89 @@ impl<G: AffineCurve> PCUniversalParams for UniversalParams<G> {
 /// polynomial.
 #[derive(Derivative)]
 #[derivative(
-    Default(bound = ""),
     Hash(bound = ""),
     Clone(bound = ""),
     Debug(bound = ""),
     Eq(bound = ""),
     PartialEq(bound = "")
 )]
 #[derive(CanonicalSerialize, CanonicalDeserialize)]
-pub struct CommitterKey<G: AffineCurve> {
-    /// The key used to commit to polynomials.
-    pub comm_key: Vec<G>,
+pub struct CommitterKey<G: Curve> {
+    /// The base point vector used for the coefficients of the polynomial.
+    pub comm_key: Vec<G::AffineRep>,
 
-    /// A random group generator.
+    /// The base point used by the opening proof to reduce to a non-blinded
+    /// opening assertion.
     pub h: G,
 
-    /// A random group generator that is to be used to make
-    /// a commitment hiding.
+    /// The base point that is to be used for hiding.
     pub s: G,
 
     /// The maximum degree supported by the parameters
     /// this key was derived from.
     pub max_degree: usize,
 
-    /// H(max_degree_comm_key, h, s, max_degree)
+    /// A fingerprint of the CommitterKey, H(max_degree_comm_key, h, s, max_degree).
+    /// Used for reference.
     pub hash: Vec<u8>,
 }
 
-impl<G: AffineCurve> CommitterKey<G> {
+impl<G: Curve> CommitterKey<G> {
     /// Scale key for testing purpose
     pub fn scale(&mut self, scaling_scalar: G::ScalarField) {
```

Comment:
Let's remove this function and put it in the test module where it's used

Reply:
This function is not used anymore in monorepo

Reply:
Either you move these changes here and report them on ginger-lib monorepo branch,
or we formally decide to address this changes in ginger-lib, but you need to create a separate branch from the monorepo branch containing them.
Currently, it's difficult to review them as in the monorepo branch everything will appear as new.

Reply:
Done